packages/dashboards/src/lib/configurator/components/widgets/table/columns-editor-v2/table-columns-configuration-v2.component.ts

```diff
@@ -0,0 +1,281 @@
+                this.changeDetector.markForCheck();
+            });
+
+        this.eventBus.subscribeUntil(DATA_SOURCE_CREATED, this.onDestroy$, (event: IEvent<IDataSource>) => {
+            if (!event.payload) {
+                return;
+            }
+
+            this.dataSource = event.payload;
+            this.changeDetector.markForCheck();
+        });
+
+        this.eventBus.subscribeUntil(DATA_SOURCE_OUTPUT, this.onDestroy$, (event: IEvent<any | IDataSourceOutput<any>>) => {
+            if (event.payload?.error) {
+                return;
+            }
+            const { dataFields } = isUndefined(event.payload.result) ? event.payload : (event.payload.result || {});
+            this.dataSourceFields = dataFields;
+
+            const disableColumnGeneration = this.dataSource?.features?.getFeatureConfig(WellKnownDataSourceFeatures.DisableTableColumnGeneration)?.enabled;
+
+            const columns = this.mergeColumns(this.dataSourceFields, this.getColumns());
+            if (columns?.length || disableColumnGeneration) {
+                // if disableColumnGeneration is enabled and there are no columns to be kept,
+                // we have to clear the form by updating the column input via the `onItemsChange` method
+                this.updateColumns(columns);
+            } else {
+                if (!disableColumnGeneration) {
+                    this.resetColumns(false);
+                }
+            }
+            this.changeDetector.markForCheck();
+        });
+        dataSourceManager.error.subscribe((err) => {
+            this.dataSourceError = err;
+            this.changeDetector.markForCheck();
+        });
+    }
+
+    public ngOnInit() {
+        this.formReady.emit(this.form);
+
+        if (this.template && !this.tableFormatterRegistry.getItems()?.length) {
+            const formatters = this.template?.[1].properties?.formatters;
+            this.tableFormatterRegistry.addItems(formatters);
+        }
+    }
+
+    public ngOnChanges(changes: SimpleChanges): void {
+        // columns from input property are set directly to the form
+        if (changes.columns) {
+            this.updateColumns(this.columns, false);
+        }
+    }
+
+    public ngOnDestroy(): void {
+        this.onDestroy$.next();
+        this.onDestroy$.complete();
+    }
+
+    public getColumns() {
+        return this.form.controls["columns"].value;
+    }
+
+    public updateColumns(columns: ITableWidgetColumnConfig[], emitEvent: boolean = true) {
+        const arr = this.form.controls["columns"] as FormArray;
+        arr.controls = columns.map(c => {
+            const fc = new FormControl(c);
+            fc.setParent(arr);
+            return fc;
+        });
+        arr.updateValueAndValidity({ emitEvent });
+
+        this.changeDetector.markForCheck();
+    }
+
+    public trackBy(index: number, item: FormControl) {
+        return item.value.id;
```

Comment:
to be safe, should we do this:
```suggestion
        return item?.value?.id;
```